refiner-lib/src/refiner_inner.rs

```diff
@@ -432,6 +439,45 @@ fn build_transaction(
                         txs.get(&hash),
                     )?
                 }
+                InnerTransactionKind::Deploy | InnerTransactionKind::DeployErc20 => {
+                    hash = virtual_receipt_id.0.try_into().unwrap();
+                    tx = tx.hash(hash).from(near_account_to_evm_address(
+                        outcome.receipt.predecessor_id.as_bytes(),
+                    ));
+
+                    tx = tx
+                        .to(None)
+                        .nonce(0)
```

Comment:
Noting for the next PR: we want to fix this so that it takes the real nonce value from the state.